File: src/main/java/money/scheduler/MoneyScheduler.java
package money.scheduler;

import java.time.LocalDate;
import java.util.List;

public interface MoneyScheduler {

    List<Integer> calculateCurrentYearSchedule(int startingMonth, int startingYear, int paymentInterval);

    List<Integer> calculateCurrentYearSchedule(LocalDate startDate, int paymentInterval);
}

File: src/main/java/money/scheduler/MoneyScheduler.kt
package money.scheduler

import java.time.LocalDate

interface MoneyScheduler {
    fun calculateCurrentYearSchedule(startingMonth: Int, startingYear: Int, paymentInterval: Int): List<Int?>
    fun calculateCurrentYearSchedule(startDate: LocalDate, paymentInterval: Int): List<Int?>
}